publication-commons/src/main/java/no/unit/nva/publication/model/business/FilesApprovalThesis.java

```diff
@@ -0,0 +1,312 @@
+import no.unit.nva.identifiers.SortableIdentifier;
+import no.unit.nva.model.Publication;
+import no.unit.nva.model.Username;
+import no.unit.nva.model.associatedartifacts.file.File;
+import no.unit.nva.model.associatedartifacts.file.PendingFile;
+import no.unit.nva.publication.model.storage.FileApprovalThesisDao;
+import no.unit.nva.publication.model.storage.TicketDao;
+import no.unit.nva.publication.service.impl.ResourceService;
+import nva.commons.core.JacocoGenerated;
+
+@SuppressWarnings({"PMD.GodClass"})
+@JsonTypeInfo(use = JsonTypeInfo.Id.NAME, property = "type")
+@JsonTypeName(FilesApprovalThesis.TYPE)
+public class FilesApprovalThesis extends TicketEntry {
+
+    public static final String TYPE = "FilesApprovalThesis";
+    @JsonProperty(IDENTIFIER_FIELD)
+    private SortableIdentifier identifier;
+    @JsonProperty(STATUS_FIELD)
+    private TicketStatus status;
+    @JsonProperty(CUSTOMER_ID_FIELD)
+    private URI customerId;
+    @JsonProperty(MODIFIED_DATE_FIELD)
+    private Instant modifiedDate;
+    @JsonProperty(CREATED_DATE_FIELD)
+    private Instant createdDate;
+    @JsonProperty(WORKFLOW)
+    private PublishingWorkflow workflow;
+    @JsonProperty(ASSIGNEE_FIELD)
+    private Username assignee;
+    @JsonProperty(OWNER_AFFILIATION_FIELD)
+    private URI ownerAffiliation;
+    @JsonProperty(APPROVED_FILES_FIELD)
+    private Set<File> approvedFiles;
+    @JsonProperty(FILES_FOR_APPROVAL_FIELD)
+    private Set<File> filesForApproval;
+
+    public FilesApprovalThesis() {
+        super();
+    }
+
+    public static FilesApprovalThesis create(Resource resource, UserInstance userInstance,
+                                             PublishingWorkflow workflow) {
+        var fileApproval = createFileApproval(resource, userInstance, workflow);
+        return REGISTRATOR_PUBLISHES_METADATA_AND_FILES.equals(workflow) ? completeAndApproveFiles(resource,
+                                                                                                   userInstance,
+                                                                                                   fileApproval)
+                   : handleMetadataOnlyWorkflow(resource, userInstance, workflow, fileApproval);
+    }
+
+    public void validateCreationRequirements(Resource resource) {
+        if (!resource.isDegree()) {
+            throw new IllegalStateException("FilesApprovalThesis requires publication to be a degree.");
+        }
+    }
+
+    @Override
+    public void validateCreationRequirements(Publication publication) {
+
+    }
+
+    public static FilesApprovalThesis createQueryObject(URI customerId, SortableIdentifier resourceIdentifier) {
+        var ticket = new FilesApprovalThesis();
+        ticket.setResourceIdentifier(resourceIdentifier);
+        ticket.setCustomerId(customerId);
+        return ticket;
+    }
+
+    @Override
+    public void validateCompletionRequirements(Publication publication) {
+
```

Comment:
Similarly

Reply:
These two validation methods are from base class and for example `validateCompletionRequirements(Publication publication)` is used in DoiRequest only and is ignored all other places. I think in general many of these overloaded method should be removed from base class :)

Will add `// NO OP` to it